packages/crypto/src/crypto.ts

```diff
@@ -48,6 +54,73 @@ export const getPublicKey = (
   return p256.getPublicKey(privateKey, isCompressed);
 };
 
+/**
+ * HPKE Encrypt Function
+ * Encrypts data using Hybrid Public Key Encryption (HPKE) standard https://datatracker.ietf.org/doc/rfc9180/.
+ *
+ * @param {HpkeEncryptParams} params - The encryption parameters including plain text, encapsulated key, and sender private key.
+ * @returns {Uint8Array} - The encrypted data.
+ */
+
+export const hpkeEncrypt = ({
+  plainText,
+  encappedKeyBuf,
+  senderPriv,
+}: HpkeEncryptParams): Uint8Array => {
+  try {
+    const senderPubBuf = getPublicKey(
+      uint8ArrayFromHexString(senderPriv),
+      false
+    );
+    const aad = buildAdditionalAssociatedData(senderPubBuf, encappedKeyBuf);
```

Comment:
A comment here here would be nice: are we making assumptions about aad structure? I think we are. It's not generic (but fine IMO). Another way to go about this would be to let callers pass their own aad if we want to maximize flexibility

Reply:
I'll add a comment here - but for now these functions still contain hardcoded values and functionalities that are specific to Turnkey's way of encrypting and decrypting. I think in a separate version we can start including modularity like passing aad, info, etc.

Reply:
Yeah makes sense to me!